aws-frauddetector-detector/src/aws_frauddetector_detector/helpers/update_worker_helpers.py

```diff
@@ -131,18 +140,37 @@ def update_detector_version_for_detector_update(afd_client,
         common_helpers.update_tags(
             frauddetector_client=afd_client,
             afd_resource_arn=latest_dv_arn,
-            new_tags=model_helpers.get_tags_from_tag_models(model.Tags)
+            new_tags=model.Tags,
         )
 
     if model.DetectorVersionStatus != DRAFT_STATUS:
+        LOG.info(
+            f"desired status is not DRAFT. updating detector version status: {model.DetectorVersionStatus}"
+        )
         api_helpers.call_update_detector_version_status(
             frauddetector_client=afd_client,
             detector_id=model.DetectorId,
             detector_version_id=max_dv_id,
             status=model.DetectorVersionStatus
         )
 
-    return {(model.DetectorId, dv_id) for dv_id in dv_ids if dv_id != max_dv_id}
+    dvs_to_delete = set()
+    new_describe_detector_response = api_helpers.call_describe_detector(
+        afd_client, model.DetectorId
+    )
+    updated_dv_summaries = new_describe_detector_response.get(
+        "detectorVersionSummaries", []
+    )
+    LOG.info(f"updated detector version summaries: {updated_dv_summaries}")
+    for summary in updated_dv_summaries:
+        dv_id = summary.get("detectorVersionId", "-1")
+        dv_status = summary.get("status", "ACTIVE")
+        if dv_id == max_dv_id or dv_status == "ACTIVE":
+            continue
+        dvs_to_delete.add((model.DetectorId, dv_id))
+
+    LOG.info(f"detector versions to delete: {dvs_to_delete}")
+    return dvs_to_delete
```

Comment:
After performing the updates to DVs, we call describe detector again to double check all active DVs, even if they're not in the previous model or current desired model (e.g. if a customer starts with `ACTIVE`, and does several updates with desired status `DRAFT`)